### PR TITLE
Add contract deployers labels

### PR DESCRIPTION
**Pull request by hildobby**
Brief comments on the purpose of your changes:

**For Dune Engine V2**

I've checked that:

### General checks:
* [ ] I tested the query on dune.com after compiling the model with dbt compile (compiled queries are written to the target directory)
* [ ] I used "refs" to reference other models in this repo and "sources" to reference raw or decoded tables 
* [ ] if adding a new model, I added a test
* [ ] the filename is unique and ends with .sql
* [ ] each sql file is a select statement and has only one view, table or function defined  
* [ ] column names are `lowercase_snake_cased`
* [ ] if adding a new model, I edited the dbt project YAML file with new directory path for both models and seeds (if applicable)
* [ ] if wanting to expose a model in the UI (Dune data explorer), I added a post-hook in the JINJA config to add metadata (blockchains, sector/project, name and contributor Dune usernames)

### Pricing checks:
* [ ] `coin_id` represents the ID of the coin on coinpaprika.com
* [ ] all the coins are active on coinpaprika.com (please remove inactive ones)

### Join logic:
* [ ] if joining to base table (i.e. ethereum transactions or traces), I looked to make it an inner join if possible

### Incremental logic:
* [ ] I used is_incremental & not is_incremental jinja block filters on both base tables and decoded tables
  * [ ] where block_time >= date_trunc("day", now() - interval '1 week')
* [ ] if joining to base table (i.e. ethereum transactions or traces), I applied join condition where block_time >= date_trunc("day", now() - interval '1 week')
* [ ] if joining to prices view, I applied join condition where minute >= date_trunc("day", now() - interval '1 week')

### CODE REVIEW

**Comment by Hosuke** [comment]
Not sure about the test failure:
https://github.com/duneanalytics/spellbook/actions/runs/4342811050/jobs/7584475224#step:9:49

File: models/labels/infrastructure/persona/contract_deployers/labels_contract_deployers.sql [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+SELECT * FROM  {{ ref('labels_contract_deployers_arbitrum') }}
+UNION
+SELECT * FROM  {{ ref('labels_contract_deployers_avalanche_c') }}
+UNION
+SELECT * FROM  {{ ref('labels_contract_deployers_bnb') }}
+UNION
+SELECT * FROM  {{ ref('labels_contract_deployers_ethereum') }}
+UNION
+SELECT * FROM  {{ ref('labels_contract_deployers_ethereum') }}
+UNION
```
Comment (Hosuke) [inline models/labels/infrastructure/persona/contract_deployers/labels_contract_deployers.sql:16]: repeated ref

File: models/labels/infrastructure/persona/contract_deployers/labels_contract_deployers_avalanche_c.sql [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+, 'Contract Deployer' as name
+, 'infrastructure' as category
+, 'hildobby' as contributor
+, 'query' AS source
+, date('2023-03-03') AS created_at
+, NOW() as modified_at
+, 'contract_deployers' AS model_name
+, 'persona' as label_type
+FROM {{ source('avalanche_c', 'creation_traces') }} creation
+LEFT ANTI JOIN {{this}} anti_table ON creation.`from`=anti_table.`from`
```
Comment (Hosuke) [inline models/labels/infrastructure/persona/contract_deployers/labels_contract_deployers_avalanche_c.sql:18]: Hi @hildobby ,
I think I know the cause of test failure.
{{this}} is not existed when initialization.
We may need to rewrite the sql logics.

**Comment by andrewhong5297** [comment]
this looks great!! @MSilb7 @chuxinh ccing you since we were chatting about this a while back, if you have any last minute ideas to add.

**Comment by MSilb7** [comment]
Hi, yeah a lot of nuance here. With this logic, I'd recommend calling it something like 'Contract Deployer EOAs.' Longer term, there may be a set of contracts creating contracts that we'd want to consider (i.e. Gnosis Safes that deploy a contract).


Related Rambling:

We did start tracking 'EOA deployed contracts' as a segment https://dune.com/oplabspbc/op-weekly-contract-deployments , but I'd been thinking about if we should expand this aperture. There's also a set of 'non-deterministic deployers' (i.e. create2 factories: https://github.com/duneanalytics/spellbook/blob/main/models/contracts/optimism/contracts_optimism_nondeterministic_contract_creators.sql

**Comment by hildobby** [comment]
@MSilb7 @andrewhong5297 were you guys looking for a slighlty different label? What exactly? These are all deployer addresses, is it specifically deployer smart contracts you want to see? I can pretty easily modify this spell to make the distinction between the two (EOA and smart contracts)

**Comment by Hosuke** [comment]
> @MSilb7 @andrewhong5297 were you guys looking for a slightly different label? What exactly? These are all deployer addresses, is it specifically deployer smart contracts you want to see? I can pretty easily modify this spell to make the distinction between the two (EOA and smart contracts)

We can merge the current PR first and start building new labels(easily modified from this spell) in another new PR.

File: models/labels/infrastructure/labels_infrastructure.sql [hunk 3]
```diff
@@ -16,6 +16,7 @@
 
  , ref('labels_flashbots_ethereum')
  , ref('labels_mev_ethereum')
+ , ref('labels_contract_deployers_ethereum')
```
Comment (jeff-dude) [inline models/labels/infrastructure/labels_infrastructure.sql:19]: assuming this should drop the `ethereum` suffix, to include all blockchains?

File: models/labels/infrastructure/persona/contract_deployers/labels_contract_deployers_arbitrum.sql [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+              , 'Contract Deployer'  AS name
+              , 'infrastructure'     AS category
+              , 'hildobby'           AS contributor
+              , 'query'              AS source
+              , date('2023-03-03')   AS created_at
+              , NOW()                AS modified_at
+              , 'contract_deployers' AS model_name
+              , 'persona'            AS label_type
+FROM {{ source('arbitrum', 'creation_traces') }} creation
+LEFT ANTI JOIN {{ source('arbitrum', 'creation_traces') }} anti_table
```
[the remark this answers is not in the document]
Comment (Hosuke) [reply models/labels/infrastructure/persona/contract_deployers/labels_contract_deployers_arbitrum.sql:22]: Sorry typed the `from` to `address` mistakenly.

[the remark this answers is not in the document]
Comment (Hosuke) [reply models/labels/infrastructure/persona/contract_deployers/labels_contract_deployers_arbitrum.sql:22]: Fixed
https://dune.com/queries/2263729?d=1

**Comment by MSilb7** [comment]
> @MSilb7 @andrewhong5297 were you guys looking for a slightly different label? What exactly? These are all deployer addresses, is it specifically deployer smart contracts you want to see? I can pretty easily modify this spell to make the distinction between the two (EOA and smart contracts)

Ah yeah I misread, this is pulling `from` from the trace, I thought it was pulling from the transaction. No changes needed

**Comment by jeff-dude** [comment]
will revisit after #3040 is set

**Comment by Hosuke** [comment]
Updated to meet new label standards in #3040

**Review by jeff-dude — APPROVED** [review]

**Pull request merged**